apps/electron/src/helper/dialog/dialog.ts

```diff
@@ -223,6 +224,24 @@ export async function loadDBFile(): Promise<LoadDBFileResult> {
       return { error: 'DB_FILE_INVALID' }; // invalid db file
     }
 
+    const db = new SqliteConnection(originalPath);
+    try {
+      await db.connect();
+      if ((await db.getMaxVersion()) === WorkspaceVersion.DatabaseV3) {
+        const tmpDBPath = await copyToTemp(originalPath);
+        await migrateToLatest(tmpDBPath, WorkspaceVersion.SubDoc);
+        originalPath = tmpDBPath;
+      }
+    } catch (error) {
+      logger.warn(
+        `loadDBFile, migration version column failed: ${originalPath}`,
+        error
+      );
+      return { error: 'DB_FILE_MIGRATION_FAILED' };
+    } finally {
+      await db.close();
+    }
+
```

Comment:
Please move all migration related codes to `../db/migration` instead.
Also, always replace the cloned db's doc rows with the migrated ones.